### PR TITLE
chore: only build modified ECS projects

### PR DESCRIPTION
**Pull request by fmiras**
Contributes to #148 

# What? 
ECS Projects only are going to build in case of modification (even at first `make build`)

**TODO:** 
- [x] Calculate stat of child folder files

### CODE REVIEW

File: scripts/buildECSprojects.ts [hunk 1]
```diff
@@ -19,10 +13,43 @@ const env = {
...
-    env,
-    stdio: 'inherit'
-  }).ref()
-})
+const folders = glob
+    .sync(path.resolve(__dirname, '../public/ecs-parcels/*/game.ts'), { absolute: true })
+    .map($ => path.dirname($))
+
+  // tslint:disable-next-line: no-floating-promises
+;(async () => {
```
Comment (menduz) [inline scripts/buildECSprojects.ts]: it is better to do 
```ts
async function main() { ... }

main().catch(err => {
  console.error(err)
  process.exit(1)
})

**Comment by eordano** [comment]
why is this blocked?

**Review by eordano — APPROVED** [review]

**Comment by fmiras** [comment]
> why is this blocked?

It was blocked because we're only stating the parents folder, that means if we change something in `game.ts` it wasn't going to compile eiter. Now that's solved

**Review by menduz — CHANGES_REQUESTED** [review]
It breaks the watch mode `node ./scripts/buildECSprojects.js --watch`

**Comment by eordano** [comment]
Is this still blocked? What's the cause of it breaking the watcher?

**Comment by fmiras** [comment]
> Is this still blocked? What's the cause of it breaking the watcher?

Because in case it doesn't have to rebuild it, it just skip `decentraland-compiler` which is the one in charge of watching those files in the first place

I'll close this PR. We should either implement this logic in `decentraland-compiler` as a `--cache` flag or implement TS 3.4 incremental builds

**Pull request closed without merging**